lib/find.ts

```diff
@@ -2,46 +2,41 @@ import * as fs from 'fs';
 import { join } from 'path';
 import { promisify } from 'util';
 import { MAX_SUPPORTED_FILE_SIZE } from './common';
-import { debug } from './debug';
 
 const readdir = promisify(fs.readdir);
 export const stat = promisify(fs.stat);
 
-export async function find(dir: string): Promise<string[]> {
-  const dirStat = await stat(dir);
-  if (!dirStat.isDirectory()) {
-    throw new Error(`${dir} is not a directory`);
-  }
-
-  const paths = await readdir(dir);
-  const absolutePaths: string[] = [];
-  for (const relativePath of paths) {
-    absolutePaths.push(join(dir, relativePath));
-  }
-
-  return await extractFilePaths(absolutePaths);
+interface FileHandler {
+  (filePath: string, stats: fs.Stats): void;
 }
 
-export async function extractFilePaths(
-  absolutePaths: string[],
-): Promise<string[]> {
+export async function find(src: string): Promise<string[]> {
   const result: string[] = [];
-  for (const absolutePath of absolutePaths) {
-    try {
-      const fileStats = await stat(absolutePath);
-      if (fileStats.isDirectory()) {
-        const subFiles = await find(absolutePath);
-        result.push(...subFiles);
-      } else if (fileStats.isFile()) {
-        if (fileStats.size > MAX_SUPPORTED_FILE_SIZE) {
-          continue;
-        }
-        result.push(absolutePath);
-      }
-    } catch (error) {
-      debug(error.message || `Error reading file ${absolutePath}. ${error}`);
+
+  await traverse(src, (filePath: string, stats: fs.Stats) => {
```

Comment:
```
  await traverse(src, (filePath: string, stats: fs.Stats) => {
    if (!stats.isFile() || stats.size >= MAX_SUPPORTED_FILE_SIZE) {
      return;
    }

    result.push(filePath);
  });
  ```
  
  Maybe all this logic could be moved to a proper function?

Reply:
I considered it yesterday, and it was my first intention. But then I realized, it's a pretty small lambda, and I would lose the possibility to collect the result easily. I would claim this is easier to read and more efficient, since there is no need to concat the result.

The condition was it's own function from the beginning, but I also decided to postpone that design since it's pretty simple (we can extract the function if needed in the future).

Reply:
I see your point and I am fine with.
The only actual concern that I am having here is that we are not handling an eventual error anywhere in find function or downstream... we are giving all the responsibility to `scan.ts` that invokes find fn to deal with that, if something that is not "size" related goes wrong we should still catch that issue and continue/try to read other files

Reply:
Yes that's also my concern, as mentioned in Slack earlier today. As promised, I'm going to delve down into the behavior, just to make sure if it's required or not.